crates/trycmd/src/runner.rs

```diff
@@ -362,7 +363,10 @@ impl Case {
 
         // For Mode::Dump's sake, allow running all
         let output = self.validate_spawn(output, step.expected_status());
-        let output = self.validate_streams(output, step, substitutions);
+        let output = match mode {
+            Mode::OnlyStatus => output,
+            _ => self.validate_streams(output, step, substitutions),
+        };
```

Comment:
The only non-trivial logic is here: we skip `validate_streams` when we have `Mode::OnlyStatus`.